tests/integration/test_constraints.py

```diff
@@ -42,29 +40,3 @@ def test_constraints(tmpdir):
     gc.save(tmpdir / 'test.pkl')
     gc = gc.load(tmpdir / 'test.pkl')
     gc.sample(10)
-
-
-_IS_VALID_CALLED = []
-
-
-def _is_valid(rows):
-    if not _IS_VALID_CALLED:
-        _IS_VALID_CALLED.append(True)
-        return pd.Series([False] * len(rows), index=rows.index)
-
-    return pd.Series([True] * len(rows), index=rows.index)
-
-
-def test_constraints_reject_sampling_zero_valid():
```

Comment:
@csala This test started failing because the first time `constraint.is_valid` is called is now within the `BaseTabularMethod.transform` method instead of when we are actually sampling data. 

I was thinking about updating the mocked `is_valid` function to return False on the second time it's called, but that seems very fragile. Instead, I moved the test for this scenario (returning zero valid rows on the first sample iteration) to a unit test in `tests/unit/tabular/test_base.py`. Let me know what you think

Reply:
Yes, sounds good!